--- openrc-0.24.2/init.d/Makefile.back	2017-04-04 16:38:29.000000000 +0300
+++ openrc-0.24.2/init.d/Makefile	2017-04-04 16:41:27.000000000 +0300
@@ -19,7 +19,7 @@
 		rc-enabled.in rpcbind.in savecore.in syslogd.in
 # These are FreeBSD specific
 SRCS-FreeBSD+=	adjkerntz.in devd.in dumpon.in encswap.in ipfw.in \
-		modules.in modules-load.in mixer.in nscd.in powerd.in syscons.in
+		modules-load.in mixer.in nscd.in powerd.in syscons.in
 
 SRCS-Linux=	agetty.in binfmt.in devfs.in dmesg.in hwclock.in consolefont.in \
 	keymaps.in killprocs.in modules.in modules-load.in mount-ro.in mtab.in \
