--- pam_passwdqc-1.0.5/Makefile.bak	2015-03-19 09:26:16.205059394 +0300
+++ pam_passwdqc-1.0.5/Makefile	2015-03-19 10:05:22.756723462 +0300
@@ -39,16 +39,10 @@
 MAP = pam_passwdqc.map
 
 all:
-	case "`uname -s`" in \
-	Linux)	$(MAKE) CFLAGS="$(CFLAGS) -DHAVE_SHADOW" \
+	case "$(USERLAND)" in \
+	GNU)	$(MAKE) CFLAGS="$(CFLAGS) -DHAVE_SHADOW" \
 			LDFLAGS="$(LDFLAGS) $(LDFLAGS_LINUX)" LDLIBS="$(LDLIBS_LINUX)" \
 			$(PROJ);; \
-	SunOS)	$(MAKE) CFLAGS="$(CFLAGS) -DHAVE_SHADOW" \
-			LD=ld LDFLAGS="$(LDFLAGS) $(LDFLAGS_SUN)" LDLIBS="$(LDLIBS_SUN)" \
-			$(PROJ);; \
-	HP-UX)	$(MAKE) CFLAGS="$(CFLAGS) -DHAVE_SHADOW" \
-			LD=ld LDFLAGS="$(LDFLAGS) $(LDFLAGS_HP)" LDLIBS="$(LDLIBS_HP)" \
-			$(PROJ);; \
 	*)	$(MAKE) LDFLAGS="$(LDFLAGS) $(LDFLAGS_GENERIC)" $(PROJ);; \
 	esac
 
